toolbox/bulk_processing/validators.py

```diff
@@ -141,7 +141,8 @@ def validate(value, **_kwargs):
 
 def region_matches_treatment_code():
     def validate(region, **kwargs):
-        if len(region.strip()) != 0 and region[0] != kwargs['row']['TREATMENT_CODE'][-1]:
+        if region.strip() and kwargs['row']['TREATMENT_CODE'].strip() and \
```

Comment:
see comment I left on other PR about maybe explaining why such defensive coding is needed